BoxContentSDK/BoxContentSDK/Operations/BOXAPIOperation.m

```diff
@@ -540,18 +540,26 @@ - (void)processResponse:(NSURLResponse *)response
 
 - (void)finishURLSessionTaskWithData:(NSData *)data response:(NSURLResponse *)response error:(NSError *)error
 {
-    if (self.HTTPResponse == nil) {
+    if (self.HTTPResponse == nil && response != nil) {
         // If the response has not already been handled (some, but not all, operations get
-        // 'processResponse:' earlier in the process) handle it here.
+        // 'processResponse:' earlier in the process), handle it here.
+        // Also, do not handle a nil response. That could happen if there was a failure on the whole
+        // network connection, for example. In that case, there is no response.
         [self processResponse:response];
     }
 
-    if (data != nil) {
+    if ([data length] > 0) {
+        // If the data object is nil (cleared out because it was already handled) or empty
+        // (the responseData property that feeds into this gets initialized as empty NSData),
+        // ignore it. Otherwise we need to process the data.
         [self processResponseData:data];
     }
+
     if (self.error == nil) {
+        // If we do not already have an error set, use the passed in error.
         self.error = error;
     }
+
```

Comment:
can we add a FIXME to review bringing this logic for if (self.error == nil) {self.error = error;} to the top of this method given the input error is coming from client, and we have expectation to preserve self.error if it was set earlier?